test/middleware.ts

```diff
@@ -1,18 +1,21 @@
 import { Middleware, ParameterizedContext, Response } from 'koa';
+import { DatasetCore } from 'rdf-js';
 import { AppContext, AppState } from '../src/app';
 
 export type Next<StateT = AppState, CustomT = AppContext> = (
   context: ParameterizedContext<StateT, CustomT>
 ) => Promise<void>;
 
+export type DatasetResponse = Response & { dataset: DatasetCore };
```

Comment:
This is a bit annoying.

Reply:
I think module augmentation might help. Here's how we extend express' Request and Response: https://github.com/zazuko/data-cube-curation/blob/master/api/typings/express/index.d.ts